python/rascaline/rascaline/utils/clebsch_gordan/_cg_cache.py

```diff
@@ -269,38 +382,43 @@ def _real2complex(lambda_: int) -> np.ndarray:
 
     See https://en.wikipedia.org/wiki/Spherical_harmonics#Real_form for details
     on the convention for how these tranformations are defined.
+
+    Operations are dispatched to the corresponding array type given by ``like``
     """
-    result = np.zeros((2 * lambda_ + 1, 2 * lambda_ + 1), dtype=np.complex128)
-    inv_sqrt_2 = 1.0 / np.sqrt(2)
-    i_sqrt_2 = 1j / np.sqrt(2)
+    result = _dispatch.zeros_like(like, shape=(2 * lambda_ + 1, 2 * lambda_ + 1))
+    inv_sqrt_2 = 1.0 / math.sqrt(2.0)
+    i_sqrt_2 = 1.0j / complex(math.sqrt(2.0))
+
     for m in range(-lambda_, lambda_ + 1):
         if m < 0:
             # Positve part
-            result[lambda_ + m, lambda_ + m] = +i_sqrt_2
+            result[lambda_ + m, lambda_ + m] = i_sqrt_2
             # Negative part
             result[lambda_ - m, lambda_ + m] = -i_sqrt_2 * ((-1) ** m)
 
         if m == 0:
-            result[lambda_, lambda_] = +1.0
+            result[lambda_, lambda_] = 1.0
 
         if m > 0:
             # Negative part
-            result[lambda_ - m, lambda_ + m] = +inv_sqrt_2
+            result[lambda_ - m, lambda_ + m] = inv_sqrt_2
             # Positive part
-            result[lambda_ + m, lambda_ + m] = +inv_sqrt_2 * ((-1) ** m)
+            result[lambda_ + m, lambda_ + m] = inv_sqrt_2 * ((-1) ** m)
 
     return result
 
 
-def _complex2real(lambda_: int) -> np.ndarray:
+def _complex2real(lambda_: int, like) -> Array:
```

Comment:
How about replacing all `lambda_` with `o3_lambda`? Since that's what is going to be used everywhere else

```suggestion
def _complex2real(o3_lambda: int, like) -> Array:
```